src/distilabel/steps/combine.py

```diff
@@ -121,3 +121,80 @@ def process(self, *inputs: StepInput) -> "StepOutput":
             merge_keys=self.inputs,
             output_merge_keys=self.outputs,
         )
+
+
+class CombineKeys(Step):
+    """Combines keys from a `StepInput`.
```

Comment:
As this short description is mainly for the component gallery and users don't necessarily know the `StepInput` I would change it to "Combine keys from a row"